docs/ADR-X-cardano-api-exports-convention.md

```diff
@@ -0,0 +1,43 @@
+# Status
+
+📜 Proposed
+
+# Context
+
+# Proposed Design
+
+## Top level exports
+
+Each top level export should export distinct symbols.
+This may depend on the case, for example: `Cardano.Api.Byron` symbols which are still in use in current eras, can be reexported through `Cardano.Api`.
+
+* `Cardano.Api` - everything domain related for the currently used and upcoming eras
```

Comment:
Unfortunately we have turned `Cardano.Api` and `Cardano.Api.Shelley` into a dumping ground.

`cardano-api/internal/Cardano/Api/*` has a mass of modules. We should start by grouping modules under sensible directories. E.g any `Serialise*.hs` module can go into a `Serialization` folder.  Eventually we can have several submodules with clear names.

 e.g `Cardano.Api.Serialization` that re-exports all of the relevant serialization modules. 

The distinction between `Cardano.Api` and `Cardano.Api.Shelley` is unclear. Because we have a split between Byron and all the other eras, I would suggest we eventually deprecate `Cardano.Api.Shelley` and use `Cardano.Api` as an "everything import" if possible. Not exporting what you mentioned below regarding utilities to avoid name clashes.